spec/unit/puppet/provider/archive/wget_spec.rb

```diff
@@ -78,5 +78,21 @@
         provider.download(name)
       end
     end
+
+    context 'proxy specified' do
+      let(:resource_properties) do
+        {
+          :name => name,
+          :source => 'http://home.lan/example.zip',
+          :proxy_type => 'https',
```

Comment:
Also with the new code we can rewrite this test without providing :proxy_type and see if it just do the right thing.